src/prefect/tasks/azureml/datastore.py

```diff
@@ -0,0 +1,256 @@
+import os
+from typing import List, Dict, Union
+
+import azureml.core.datastore
+from azureml.core.workspace import Workspace
+from azureml.data.azure_storage_datastore import (
+    AzureBlobDatastore,
+    AbstractAzureStorageDatastore,
+)
+from azureml.data.data_reference import DataReference
+
+from prefect import Task
+from prefect.client import Secret
+from prefect.utilities.tasks import defaults_from_attrs
+
+
+class DatastoreRegisterBlobContainer(Task):
+    """
+    Task for registering Azure Blob Storage container as a Datastore in a Azure ML service Workspace. 
+
+    Args:
+        - workspace (azureml.core.workspace.Workspace): The Workspace to which the Datastore is to be registered.
+        - container_name (str, optional): The name of the container.
+        - datastore_name (str, optional): The name of the datastore. If not defined, the container name will be used.
+        - create_container_if_not_exists (bool, optional): Create a container, if one does not exist with the given name. 
+        - overwrite_existing_datastore (bool, optional): Overwrite an existing datastore. If the datastore does not exist, it will be created.
+        - azure_credentials_secret (str, optinonal): The name of the Prefect Secret that stores your Azure credentials; this Secret must be a JSON string with two keys: `ACCOUNT_NAME` and `ACCOUNT_KEY`
```

Comment:
Is this the only task that requires azure credentials?

Reply:
The authentication to the Azure ML service is done through the [`Workspace`-object](https://docs.microsoft.com/en-us/python/api/azureml-core/azureml.core.workspace.workspace?view=azure-ml-py). 

In the case you point out, we are registering/linking a Blob Container to the `Workspace`, and the we're authenticating to the Blob storage, not the Workspace. 

The authentication in Blob storage is different from Azure ML service (and managing resources in Azure more generally). See [this Jupypter notebook](https://github.com/Azure/MachineLearningNotebooks/blob/master/how-to-use-azureml/manage-azureml-service/authentication-in-azureml/authentication-in-azureml.ipynb) for an example.

Reply:
Okay that makes a lot more sense, thanks for explaining and linking the notebook!